ui/helpers/utils/util.js

```diff
@@ -87,17 +86,77 @@ export function addressSummary(
     : '...';
 }
 
-export function isValidDomainName(address) {
-  const match = punycode
-    .toASCII(address)
-    .toLowerCase()
-    // Checks that the domain consists of at least one valid domain pieces separated by periods, followed by a tld
-    // Each piece of domain name has only the characters a-z, 0-9, and a hyphen (but not at the start or end of chunk)
-    // A chunk has minimum length of 1, but minimum tld is set to 2 for now (no 1-character tlds exist yet)
-    .match(
-      /^(?:[a-z0-9](?:[-a-z0-9]*[a-z0-9])?\.)+[a-z0-9][-a-z0-9]*[a-z0-9]$/u,
-    );
-  return match !== null;
+export function isValidENSDomainName(address) {
+  let result = false;
+  const tlds = ['.test', '.eth'];
+  tlds.forEach((tld) => {
+    if (address.toLowerCase().endsWith(tld)) {
+      result = true;
+    }
+  });
+  return result;
+}
+
+export async function isValidUnstoppableDomainName(address, inputTlds) {
+  let tlds;
+  if (inputTlds) {
+    tlds = inputTlds;
+  } else {
+    tlds = await getUdTlds();
+  }
+  let result = false;
+  tlds.forEach((tld) => {
+    if (address.toLowerCase().endsWith(`.${tld}`)) {
+      result = true;
+    }
+  });
+  return result;
+}
+export async function getUdTlds() {
+  let result = [];
+  try {
+    const url = 'https://resolve.unstoppabledomains.com/supported_tlds';
+    const response = await fetch(url);
+    const data = await response.json();
+    result = data.tlds;
+    return result;
+  } catch {
+    return result;
+  }
+}
+
+export async function getAndParseUdCurrencies() {
+  const resultObject = {
+    singleChain: [],
+    multiChain: [],
+  };
+  try {
+    const url = 'https://unstoppabledomains.com/api/uns-resolver-keys';
+    const response = await fetch(url);
+    const data = await response.text();
+    const currencyArray = parseKeysArray(JSON.parse(data).keys);
+    currencyArray.forEach(function (crypto) {
+      if (crypto.includes('.version.')) {
+        resultObject.multiChain.push(crypto.split('.')[1]);
+      } else {
+        resultObject.singleChain.push(crypto.split('.')[1]);
+      }
+    });
+    return resultObject;
+  } catch {
+    return resultObject;
+  }
+}
```

Comment:
These two methods are adding centralized services to our code, and exposing our user's ip addresses to unstoppable domains. Is there any alternative to http requests to retrieve this info? @Gudahtt can I get your take on this as well?

Reply:
As of now, our only alternative to this would be to hardcode constants for the TLDs and Supported Currencies - so each time those change, an update to the codebase would be required

Reply:
if it helps, all of the data contained within the API is open source which is less centralized